### PR TITLE
Fix bugs in edit command messages

### LINKED ISSUE #231: [PE-D][Tester D] [UG] Behaviour of Edit After Filter not Clear in Error Message and UG
Running "edit m/transactions ..." after "filter buy" results in an error message stating that "Edit of transaction can only happen when transactions are visible in the application! Use 'view' command to view a specific client before applying this command". 

Although the second sentence of the error message is correct, the first sentence is not technically right, since the transactions are visible, but not editable. Since the criteria is that "view" must explicitly be used, not that the transactions are visible, the UG can also be updated to clarify this behaviour.

-------------
Labels: `type.DocumentationBug` `severity.Medium`
original: junweimoo/ped#4

### PR DESCRIPTION
**Pull request by notle1706**

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/EditTransactionCommand.java [hunk 1]
```diff
@@ -60,7 +59,7 @@ public CommandResult execute(Model model) throws CommandException {
         TransactionLog transactionLog = currentClient.getTransactions();
 
         if (index.getZeroBased() >= transactionLog.size()) {
-            throw new CommandException(Messages.MESSAGE_INVALID_CLIENT_DISPLAYED_INDEX);
+            throw new CommandException(Messages.MESSAGE_INVALID_TRANSACTION_DISPLAYED_INDEX);
```
Comment (quannam0124) [inline src/main/java/seedu/address/logic/commands/EditTransactionCommand.java:62]: Good job discovering this bug! LGTM!

**Review by quannam0124 — APPROVED** [review]
LGTM Elton!

File: src/main/java/seedu/address/logic/commands/EditTransactionCommand.java (hunk 1 above)
Comment (jicsontoh) [reply src/main/java/seedu/address/logic/commands/EditTransactionCommand.java:62]: Good catch!

**Review by jicsontoh — APPROVED** [review]
LGTM

**Pull request merged**